### DISCUSSION TITLE
owner of a repo that is a submodule

### DISCUSSION THREAD

**Discussion by extrawurst** [libgit2/libgit2#6194] [Q&A]
If I use `git_repository_open_ext` on a path that is a submodule it will correctly discover the `.git`-path inside the parents `.git/modules/...`. 
Is there a way to programmatically detect that the resulting `git_repository *` is a submodule and where to find its owner? I can't seem to find anything in the API docs.

**Comment by ethomson** [comment]
I don't think so.  In fact I don't *think* that there's a guarantee that a repository with an external git folder (and a `.git` file pointing to it) is a submodule but I could be mistaken.  Submodules are definitely not my specialty.

**Comment by boretrk** [comment]
I might be wrong here but I think the way git does it is by going to GIT_DIR, stepping up to the parent directory and scan for a repository from there.
Once found it checks if the outer repository has a submodule that matches the expected worktree. (To make sure that it's not just a repo that is checked into another repo.)

**Reply by extrawurst** [reply]
@boretrk will go with your implementation idea
